### PR TITLE
Added lcc

### PR DESCRIPTION
**Pull request by cnegre**
Adding LCC package

### CODE REVIEW

File: var/spack/repos/builtin/packages/lcc/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    lattice. Provided you have a ‘.pdb‘ file containing your lattice basis you can
+    create a solid or slab from command line."""
+
+    homepage = "https://github.com/lanl/LCC.git"
+    url = "https://github.com/lanl/LCC/archive/refs/tags/v1.0.1.tar.gz"
+    git = "https://github.com/lanl/LCC.git"
+
+    maintainers("cnegre")
+
+    version("main", branch="main", prefered=True)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/lcc/package.py]:
```suggestion
    version("main", branch="main", preferred=True)
```

File: var/spack/repos/builtin/packages/lcc/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    depends_on("qmd-progress")
+
+    root_cmakelists_dir = "src"
+
+    def cmake_args(self):
+        spec = self.spec
+        args = ["-DCMAKE_Fortran_FLAGS=-ffree-line-length-none"]
+        args.append("-DCMAKE_Fortran_FLAGS=-g -fopenmp")
+
+        if "+shared" in spec:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/lcc/package.py]: This should correspond to a variant.  See https://spack.readthedocs.io/en/latest/packaging_guide.html#boolean-variants.

File: var/spack/repos/builtin/packages/lcc/package.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            args.append("-DBUILD_SHARED_LIBS=ON")
+        else:
+            args.append("-DBUILD_SHARED_LIBS=OFF")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/lcc/package.py]: There are also some helpers that can simplify this (e.g., `self.define_from_variant`), see https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
There's a typo and a couple of suggestions/fixes.

**Review by tldahlgren — APPROVED** [review]
Confirmed the branch exists and the two version sha256 are correct.  Attempt to install seemed to take too long but `spack spec` succeeded.

**Review by tldahlgren — DISMISSED** [review]
Looks like a bunch of unrelated commits were pulled into this PR.  Please drop the unrelated commits.

File: var/spack/repos/builtin/packages/lcc/package.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    maintainers("cnegre")
+
+    version("main", branch="main", preferred=True)
+    version("1.0.1", sha256="fa13364dcdf3b1f8d80fc768f0e7ad3849f8d98091fb96926100a6764f836020")
+    version("1.0.0", sha256="750ce09e809a4e85ae3219fd537dc84a923fe3d3683b26b5d915eccfd1f0120c")
+
+    variant("shared", default=False, description="Build shared libs")
+
+    depends_on("cmake@3.10:", type="build")
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/lcc/package.py:27]: @cnegre: note that I updated this per your `CMakeLists.txt` file

File: var/spack/repos/builtin/packages/lcc/package.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    homepage = "https://github.com/lanl/LCC.git"
+    url = "https://github.com/lanl/LCC/archive/refs/tags/v1.0.1.tar.gz"
+    git = "https://github.com/lanl/LCC.git"
+
+    maintainers("cnegre")
+
+    version("main", branch="main", preferred=True)
+    version("1.0.1", sha256="fa13364dcdf3b1f8d80fc768f0e7ad3849f8d98091fb96926100a6764f836020")
+    version("1.0.0", sha256="750ce09e809a4e85ae3219fd537dc84a923fe3d3683b26b5d915eccfd1f0120c")
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/lcc/package.py:23]: Also it's not needed for this PR, but you should make stable tarballs for these (and call `v1.0.1` `v1.0.1`, not `cmake` as it says in your releases).  See https://github.blog/2023-02-21-update-on-the-future-stability-of-source-code-archives-and-hashes/

**Comment by tgamblin** [comment]
@spackbot run pipeline

**Review by tldahlgren — APPROVED** [review]
Re-confirmed I can successfully build `lcc@main` (with Todd's changes).

**Pull request merged**